### ISSUE TITLE
Improve controller input latency

### ISSUE THREAD

**Issue by antangelo** [xemu-project/xemu#746]
### Feature Request

I made a relatively crude measurement of input latency using 5451001a by recording an input at 60fps and playing it back in slow motion.

https://user-images.githubusercontent.com/25046221/155468487-ab7aaae1-d225-419d-981f-1f11e3dc3e24.mp4

Going off of the frame counter, it's about 6-7 frames between the input being recognized by the controller input window, and the input being recognized by the game.

I intend to come up with a more rigorous/automated method of testing input latency to supplement this.

**Comment by GXTX** [comment]
Would be interested in seeing what sort of latency is there when directly passing the controller to the guest, also perhaps it's not input but rather display.

**Comment by dracc** [comment]
Please consider doing the same test on physical hardware too, maybe the game is slow?

**Comment by antangelo** [comment]
> Please consider doing the same test on physical hardware too, maybe the game is slow?

I agree that testing on real hardware would be ideal. It's a bit difficult to perform the same test on hardware given that it's not as easy to read the controller state independent from the game (without specialized hardware to do so). Is the nxdk controller sample known to have low latency? I can perform the above test on that for now to isolate the game as potentially being slow.

**Comment by antangelo** [comment]
Using the nxdk sample, the latency seems to be 2-3 frames. It's also possible part of the latency is in the display as GXTX mentioned, which could impact these results.

https://user-images.githubusercontent.com/25046221/155479913-fe801aaf-9eca-4a6c-b951-14a60b0bcf85.mp4

I'll see if I can devise a hardware testing method for both of these to get some measurements to compare against.

**Comment by Spidy123222** [comment]
> Using the nxdk sample, the latency seems to be 2-3 frames. It's also possible part of the latency is in the display as GXTX mentioned, which could impact these results.
> 
> 
> 
> https://user-images.githubusercontent.com/25046221/155479913-fe801aaf-9eca-4a6c-b951-14a60b0bcf85.mp4
> 
> 
> 
> I'll see if I can devise a hardware testing method for both of these to get some measurements to compare against.

I would think to get a equal grounds for reading with hardware and emulator you would need a fast camera and a led connected to  a button on controller to know when you truely activated the button. Then can be cross referenced.
